inference/core/workflows/core_steps/models/foundation/florence2/v1.py

```diff
@@ -0,0 +1,222 @@
+    images: Union[WorkflowImageSelector, StepOutputImageSelector] = ImageInputField
+    task_type: TaskType = Field(
+        description="Task type to be performed by model. Value of parameter determine set of fields "
+        "that are required. For `unconstrained`, `visual-question-answering`, "
+        " - `prompt` parameter must be provided."
+        "For `structured-answering` - `output-structure` must be provided. For "
+        "`classification`, `multi-label-classification` and `object-detection` - "
+        "`classes` must be filled. `ocr`, `caption`, `detailed-caption` do not"
+        "require any additional parameter.",
+    )
+    prompt: Optional[Union[WorkflowParameterSelector(kind=[STRING_KIND]), str]] = Field(
+        default=None,
+        description="Text prompt to the Claude model",
+        examples=["my prompt", "$inputs.prompt"],
+        json_schema_extra={
+            "relevant_for": {
+                "task_type": {"values": TASKS_REQUIRING_PROMPT, "required": True},
+            },
+        },
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+
+    @model_validator(mode="after")
+    def validate(self) -> "BlockManifest":
+        if self.task_type in TASKS_REQUIRING_PROMPT and self.prompt is None:
+            raise ValueError(
+                f"`prompt` parameter required to be set for task `{self.task_type}`"
+            )
+        return self
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name="output", kind=[STRING_KIND, LANGUAGE_MODEL_OUTPUT_KIND]
+            )
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class Florence2BlockV1(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+        step_execution_mode: StepExecutionMode,
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+        self._step_execution_mode = step_execution_mode
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key", "step_execution_mode"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        images: Batch[WorkflowImageData],
+        task_type: TaskType,
+        prompt: Optional[str],
+        model_version: str,
+    ) -> BlockResult:
+        if self._step_execution_mode is StepExecutionMode.LOCAL:
+            return self.run_locally(
+                images=images,
+                task_type=task_type,
+                model_version=model_version,
+                prompt=prompt,
+            )
+        elif self._step_execution_mode is StepExecutionMode.REMOTE:
+            raise NotImplementedError(
```

Comment:
just exploratory question - am I correct that for base model we could run it in core models lambda given special endpoint is created?

Reply:
Yes, I believe so. We'd need to enable it with envvar also. It might be pretty dang slow depending on the task.

Reply:
yeah, true that